app/models/v2/test_result.rb

```diff
@@ -70,6 +70,15 @@ class TestResult < ApplicationRecord
       { conditions: systems.arel.where_sql.gsub(/^where /i, '') }
     end
 
+    searchable_by :failed_rule_severity, %i[eq in] do |_key, op, val|
+      values = ['IN'].include?(op) ? val.split : val
```

Comment:
You don't need to deal with the operator at all, the `split` returns a single element array when no separator is detected. So you can always call `split`, see other existing examples.